### PR TITLE
Instant upload in the context of multi account support

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Work in progress PR for #25

### CODE REVIEW

File: src/com/owncloud/android/ui/activity/Preferences.java [hunk 1]
```diff
@@ -560,23 +566,33 @@ private AppCompatDelegate getDelegate() {
...
     /**
-     * Save the "Instant Upload Path" on preferences
+     * Save the "Instant Upload Path" and corresponding account on preferences
      */
     private void saveInstantUploadPathOnPreferences() {
         SharedPreferences appPrefs =
                 PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
         SharedPreferences.Editor editor = appPrefs.edit();
         editor.putString("instant_upload_path", mUploadPath);
+        editor.putString("instant_upload_path_account", mUploadPathAccount);
```
Comment (przybylski) [inline src/com/owncloud/android/ui/activity/Preferences.java]: putString returns Editor so you can chain but thats nothing big

File: src/com/owncloud/android/ui/activity/Preferences.java [hunk 2]
```diff
@@ -587,17 +603,20 @@ private void loadInstantUploadVideoPath() {
         SharedPreferences appPrefs =
                 PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
         mUploadVideoPath = appPrefs.getString("instant_video_upload_path", getString(R.string.instant_upload_path));
-        mPrefInstantVideoUploadPath.setSummary(mUploadVideoPath);
+        mUploadVideoPathAccount = appPrefs.getString("instant_video_upload_path_account",
```
Comment (przybylski) [inline src/com/owncloud/android/ui/activity/Preferences.java]: Could you please use constant for preference name

**Comment by AndyScherzinger** [comment]
Instant Upload is brokoen on Android N, can some else test if this is working? pinging @przybylski @tobiasKaminsky @Spacefish :)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky imho obsolete as soon as #358 is done but keeping it open for now. (Right?)

**Comment by tobiasKaminsky** [comment]
Is this something still up to date as of the big rewrite of @mario?

**Comment by mario** [comment]
I have to check what is this.

**Comment by mario** [comment]
Looks outdated, can be deleted.

I think we need to respect account disabled for Auto Upload, but dont think
this is something that is respected across the app at all atm, and the rest
of this is useless since instant upload disappears.

**Comment by AndyScherzinger** [comment]
This one is obsolete as soon as auto upload is active for all Android versions!
Thanks to @mario's efforts with auto upload 🚀

**Comment by tobiasKaminsky** [comment]
> This one is obsolete as soon as auto upload is active for all Android versions!

I thought that auto Upload is only possible for >= 5 as of JobScheduler?

**Comment by AndyScherzinger** [comment]
AFAIK it'll work on earlier versions too since @mario integrated Evernote job library which will fallback to an AlarmManager implementation to mimic JobScheduler so that should then be solved too. Right @mario?

**Comment by mario** [comment]
It will use GCM network manager on 4 starting Nextcloud 1.5 :)

1.4.2 enables it only for 6 and 7.

**Comment by tobiasKaminsky** [comment]
Thanks for explanation --> closing

**Pull request closed without merging**